crates/core/tedge_agent/src/entity_manager/server.rs

```diff
@@ -211,6 +211,23 @@ impl EntityStoreServer {
             }
         }
 
+        if channel.is_alarm()
+            || channel.is_command()
+            || channel.is_command_metadata()
+            || channel.is_health()
```

Comment:
We're only clearing the retained messages on this specified list of channels where we expect retained messages, excluding measurement, event and any other random channels under the entity topic id. I'm open to clear any measurements and events that the user may have accidentally published as retained. But, for other random channels, I feel we shouldn't be touching them, as tedge doesn't recognise/own those topics.

Reply:
That makes sense.